src/components/GraphiQL.js

```diff
@@ -741,6 +749,23 @@ export class GraphiQL extends React.Component {
     }
   });
 
+  handleCopyQuery = () => {
+    const editor = this.getQueryEditor();
+    const query = editor.getValue();
+
+    if (!query) {
+      return;
+    }
+
+    const formattedQuery = print(parse(query));
```

Comment:
Since the user can click prettify and then copy easily, and since prettify currently drops comments, I suggest you just copy the query without formatting.